app/controllers/graphql_controller.rb

```diff
@@ -13,6 +13,7 @@ def execute
       # current_user: current_user,
     }
     result = NextGenAuthAppDemoSchema.execute(query, variables: variables, context: context, operation_name: operation_name)
+    response.set_header('X-Shopify-Request-Auth-Code', 'true') if result.to_h["data"]["testField"]["errors"].present?
```

Comment:
1. This header is currently being set by the graphql_controller. Do we want to abstract this responsibility to a middleware of some sorts?
2. We had a discussion on setting the header `Access-Control-Expose-Headers: X-Shopify-Request-Auth-Code` on all top-level responses. This would be needed if the app uses CORS: https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Access-Control-Expose-Headers#:~:text=The%20Access%2DControl%2DExpose%2D,Content%2DLanguage

Any thoughts on this?

Reply:
This header should be set on all responses. We could have the middleware set it to false by default.